include/boost/geometry/algorithms/detail/envelope/implementation.hpp

```diff
@@ -43,27 +46,31 @@ namespace detail { namespace envelope
 {
 
 
-struct exterior_ring_expand_policy
-{
-    template <typename Box, typename Geometry>
-    static inline void apply(Box& mbr, Geometry const& geometry)
-    {
-        Box ring_mbr;
-        envelope_range<>::apply(exterior_ring(geometry), ring_mbr);
-        geometry::expand(mbr, ring_mbr);
-    }
-};
-
-
+template <std::size_t Dimension, std::size_t DimensionCount>
```

Comment:
I am afraid do it the way you suggest would greatly increase the code size.
I thought that dispatching with respect to both dimension and coordinate system is not that complicated. In `bg::distance()` we already dispatch over multiple types at the same time...